pandas/tests/series/test_rank.py

```diff
@@ -9,441 +9,449 @@
-        exp[mask] = np.nan
-
-        tm.assert_series_equal(ranks, exp)
-
-        iseries = Series(np.arange(5).repeat(2))
-
-        iranks = iseries.rank()
-        exp = iseries.astype(float).rank()
-        assert_series_equal(iranks, exp)
-        iseries = Series(np.arange(5)) + 1.0
-        exp = iseries / 5.0
-        iranks = iseries.rank(pct=True)
-
-        assert_series_equal(iranks, exp)
-
-        iseries = Series(np.repeat(1, 100))
-        exp = Series(np.repeat(0.505, 100))
-        iranks = iseries.rank(pct=True)
-        assert_series_equal(iranks, exp)
-
-        iseries[1] = np.nan
-        exp = Series(np.repeat(50.0 / 99.0, 100))
-        exp[1] = np.nan
-        iranks = iseries.rank(pct=True)
-        assert_series_equal(iranks, exp)
-
-        iseries = Series(np.arange(5)) + 1.0
-        iseries[4] = np.nan
-        exp = iseries / 4.0
-        iranks = iseries.rank(pct=True)
-        assert_series_equal(iranks, exp)
-
-        iseries = Series(np.repeat(np.nan, 100))
-        exp = iseries.copy()
-        iranks = iseries.rank(pct=True)
-        assert_series_equal(iranks, exp)
-
-        iseries = Series(np.arange(5)) + 1
-        iseries[4] = np.nan
-        exp = iseries / 4.0
-        iranks = iseries.rank(pct=True)
-        assert_series_equal(iranks, exp)
-
-        rng = date_range("1/1/1990", periods=5)
-        iseries = Series(np.arange(5), rng) + 1
-        iseries.iloc[4] = np.nan
-        exp = iseries / 4.0
-        iranks = iseries.rank(pct=True)
-        assert_series_equal(iranks, exp)
-
-        iseries = Series([1e-50, 1e-100, 1e-20, 1e-2, 1e-20 + 1e-30, 1e-1])
-        exp = Series([2, 1, 3, 5, 4, 6.0])
-        iranks = iseries.rank()
-        assert_series_equal(iranks, exp)
-
-        # GH 5968
-        iseries = Series(["3 day", "1 day 10m", "-2 day", NaT], dtype="m8[ns]")
-        exp = Series([3, 2, 1, np.nan])
-        iranks = iseries.rank()
-        assert_series_equal(iranks, exp)
-
-        values = np.array(
-            [-50, -1, -1e-20, -1e-25, -1e-50, 0, 1e-40, 1e-20, 1e-10, 2, 40],
-            dtype="float64",
-        )
-        random_order = np.random.permutation(len(values))
-        iseries = Series(values[random_order])
-        exp = Series(random_order + 1.0, dtype="float64")
-        iranks = iseries.rank()
-        assert_series_equal(iranks, exp)
-
-    def test_rank_categorical(self):
-        # GH issue #15420 rank incorrectly orders ordered categories
-
-        # Test ascending/descending ranking for ordered categoricals
-        exp = Series([1.0, 2.0, 3.0, 4.0, 5.0, 6.0])
-        exp_desc = Series([6.0, 5.0, 4.0, 3.0, 2.0, 1.0])
-        ordered = Series(
-            ["first", "second", "third", "fourth", "fifth", "sixth"]
-        ).astype(
-            CategoricalDtype(
-                categories=["first", "second", "third", "fourth", "fifth", "sixth"],
-                ordered=True,
-            )
+    return _results
```

Comment:
Not sure if these fixtures should rather go into `conftest.py` - but since they're only used in this module, I chose to put them here for now

Reply:
only general purpose ones
will look in a little